### ISSUE TITLE
Can't set download directory for book from OPDS catalog

### ISSUE THREAD

**Issue by asampal** [koreader/koreader#2441]
* KOReader version: g1a83eb5
* Device: Kobo Aura One FW 4.2.8110

#### Issue
Can't set the download directory for a book to be downloaded from local Calibre OPDS catalog

#### Steps to reproduce
- Start content server on Calibre 2.74
- In KOReader's File Manager, with wifi on, choose OPDS Catalog/Local calibre catalog
- Browse catalog "By Title" and choose a book
- See prompt for setting download directory and choose that option
- In the resulting directory listing, choose a subdirectory (under /mnt/onboard) by tap-holding on a name
- At the resulting confirmation prompt, choose "Confirm"
- Find yourself at the title listing in the catalog with the prompt for setting the download directory still showing, as if nothing happened (expected some progress indicator that book is being downloaded)
- Return to the file manager and check directory set for downloading - nothing is there

**Comment by KenMaltby** [comment]
When you come back to "the title listing in the catalog with the prompt for setting the download directory still showing" it should be showing the selection(s) for file format to download, if you click on the one you want it should download the file.  You should also see an info box that states where the file is being downloaded to.  For me it is always to the folder I had set, and the file can be found there.

**Comment by asampal** [comment]
@KenMaltby your comment put me on to my (user) error. The prompt for setting the download directory is a two part popup - the top part shows a single word reflecting the file type (PDF, in my case) while the bottom has the text "Set download directory".  It wasn't obvious to me that the top part was a selectable option - I thought it was simply informational, showing me the type of the title I had just selected in the catalog. Once I tapped that, I got the download, as expected.

From a UX point of view, it might help if the top of this prompt would have text like "Download EPUB | Download PDF" to better indicate that it is an option for initiating an action.

**Comment by houqp** [comment]
@asampal I like your suggestion, do you mind sending a PR to fix the wording?

**Comment by asampal** [comment]
[Here](https://github.com/koreader/koreader/pull/2445) you go.

**Comment by asampal** [comment]
Looking at the source, I see that there is a possibility for there to be multiple formats for one selected title, in which case each button having "Download " prefixed might be redundant. It might make more sense for that case to have a title bar (if that can be done), with the title of "Download" and to leave the button for each format with only the text of the format.

**Comment by houqp** [comment]
I agree, do you want to update your PR? If so, I can help point you to the right direction.

**Comment by asampal** [comment]
NP, I'll get to it soon.

**Comment by KenMaltby** [comment]
The local Calibre OPDS catalog is a good example, where ebooks may be acquired in many formats and converted to a common one that your device supports well.   Also an online book source may offer ebooks in several formats, in their catalog listings.

**Comment by asampal** [comment]
I started working on a more general solution for this, but got stuck with refreshing a dialog box after returning from the one where you choose a new directory to download to. Specifically, I wanted the title of the initial dialog to contain wording like "Choose a download format to download to <path>". Not being familiar with the GUI stuff, I couldn't get a new path to be updated. 

I'll take another look and report what I tried and maybe someone here can give me some pointers.
